src/test/java/org/inferred/freebuilder/processor/property/BiMapPropertyTest.java

```diff
@@ -0,0 +1,638 @@
+            .addLine("  .putItems(%s, %s)", keys.example(1), values.example(1))
+            .addLine("  .build();")
+            .addLine("assertThat(value.%s).isEqualTo(%s);",
+                convention.get(), exampleBiMap(0, 0, 1, 1)
+            )
+            .build())
+        .runTest();
+  }
+
+  @Test
+  public void testOverridingAdd() {
+    behaviorTester
+        .with(SourceBuilder.forTesting()
+            .addLine("package com.example;")
+            .addLine("@%s", FreeBuilder.class)
+            .addLine("public interface DataType {")
+            .addLine("  %s<%s, %s> %s;", BiMap.class, keys.type(), values.type(), convention.get())
+            .addLine("")
+            .addLine("  class Builder extends DataType_Builder {")
+            .addLine("  @Override public Builder putItems(%s key, %s value) {",
+                keys.unwrappedType(), values.unwrappedType()
+            )
+            .addLine("    return this;")
+            .addLine("  }")
+            .addLine("  }")
+            .addLine("}"))
+        .with(testBuilder()
+            .addLine("DataType value = new DataType.Builder()")
+            .addLine("  .putItems(%s, %s)", keys.example(0), values.example(0))
+            .addLine("  .putItems(%s, %s)", keys.example(1), values.example(1))
+            .addLine("  .putAllItems(%s)", exampleBiMap(2, 2, 3, 3))
+            .addLine("  .build();")
+            .addLine("assertThat(value.%s).isEmpty();", convention.get())
+            .build())
+        .runTest();
+  }
+
+  @Test
+  public void testEquality() {
+    behaviorTester
+        .with(bimapPropertyType)
+        .with(testBuilder()
+            .addLine("new %s()", EqualsTester.class)
+            .addLine("  .addEqualityGroup(")
+            .addLine("    new DataType.Builder().build(),")
+            .addLine("    new DataType.Builder().build())")
+            .addLine("  .addEqualityGroup(")
+            .addLine("    new DataType.Builder()")
+            .addLine("      .putItems(%s, %s)", keys.example(0), values.example(0))
+            .addLine("      .build(),")
+            .addLine("    new DataType.Builder()")
+            .addLine("      .putItems(%s, %s)", keys.example(0), values.example(0))
+            .addLine("      .build())")
+            .addLine("  .addEqualityGroup(")
+            .addLine("    new DataType.Builder()")
+            .addLine("      .putItems(%s, %s)", keys.example(0), values.example(0))
+            .addLine("      .putItems(%s, %s)", keys.example(1), values.example(1))
+            .addLine("      .build(),")
+            .addLine("    new DataType.Builder()")
+            .addLine("      .putItems(%s, %s)", keys.example(0), values.example(0))
+            .addLine("      .putItems(%s, %s)", keys.example(1), values.example(1))
+            .addLine("      .build())")
+            .addLine("  .testEquals();")
+            .build())
+        .runTest();
+  }
+
+  @Test
+  @Ignore
```

Comment:
I didn't realize it, but BiMaps aren't supported by jackson https://github.com/FasterXML/jackson-datatypes-collections/blob/master/guava/src/main/java/com/fasterxml/jackson/datatype/guava/GuavaDeserializers.java#L170. Although ImmutableBiMaps are.

Reply:
Could you put that explanation in the Ignore annotation pls

Reply:
Done